data_science/archive_exploration/01 - subject coocurrence.ipynb

```diff
@@ -0,0 +1,320 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Archive data\n",
+    "The Wellcome archive sits in a collections management system called CALM, which follows a rough set of standards and guidelines for storing archival records called [ISAD(G)](https://en.wikipedia.org/wiki/ISAD(G). The archive is comprised of _collections_, each of which has a hierarcal set of series, sections, subjects, items and pieces sitting underneath it.  \n",
+    "In the following notebooks I'm going to explore it and try to make as much sense of it as I can programatically.\n",
+    "\n",
+    "Let's start by loading in a few useful packages and defining some nice utils."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%matplotlib inline\n",
+    "import matplotlib.pyplot as plt\n",
+    "import seaborn as sns\n",
+    "sns.set_style('white')\n",
+    "plt.rcParams['figure.figsize'] = (20, 20)\n",
+    "\n",
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "import networkx as nx\n",
+    "\n",
+    "from umap import UMAP\n",
+    "from tqdm import tqdm_notebook as tqdm"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def flatten(list):\n",
+    "    return [item for sublist in list for item in sublist]\n",
+    "\n",
+    "def cartesian(*arrays):\n",
+    "    return np.array([x.reshape(-1) for x in np.meshgrid(*arrays)]).T\n",
+    "\n",
+    "def clean_subject(subject):\n",
+    "    return subject.strip().lower().replace('<p>', '')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "let's load up our CALM data (exported in its entirity as a single `.json`, where each line is a record)"
```

Comment:
Worth including a brief note on how somebody could get that export if they don’t have it already?